brave/src/main/java/brave/propagation/B3SingleFormat.java

```diff
@@ -251,69 +259,48 @@ public static TraceContextOrSamplingFlags parseB3SingleFormat(CharSequence b3, i
   }
 
   /** Returns zero if truncated, malformed, or too big after logging */
-  static long tryParseParentId(CharSequence b3, int pos, int endIndex) {
-    if (endIndex < pos + 16) {
-      Platform.get().log("Invalid input: truncated", null);
+  static long tryParseId(String name, CharSequence value, int beginIndex, int endIndex) {
+    int endOfId = beginIndex + 16;
+    if (beginIndex == endIndex || value.charAt(beginIndex) == '-') {
```

Comment:
Not sure if this method is used elsewhere - but at a glance it seems this duplicates some validation with the validation code in `parseB3SingleFormat`. Maybe fine to be safer than sorry though.

Reply:
traceID is a special case, as it can be variable length, and the sampled flag is single-character.. I couldn't re-use this logic for all field permutations, but most of them I could!

Reply:
I loosened one log entry to reduce the logic with little loss in fidelity. PTAL!